gui/gui/src/TGTab.cxx

```diff
@@ -129,37 +131,59 @@ void TGTabElement::DrawBorder()
 
 ////////////////////////////////////////////////////////////////////////////////
 /// Handle button event in the tab widget. Basically we only handle
-/// button events in the small tabs.
+/// button and scroll events in the small tabs.
 
 Bool_t TGTabElement::HandleButton(Event_t *event)
 {
    if (event->fType == kButtonPress) {
-      TGTab* main = (TGTab*)fParent;
-      if (main) {
-         if (fShowClose && event->fWindow == GetId() &&
-             (UInt_t)event->fX > fTWidth+12 && (UInt_t)event->fX < fTWidth+26 &&
-             (UInt_t)event->fY > fHeight/2-7 && (UInt_t)event->fY < fHeight/2+7) {
-            if (main->GetTabTab(main->GetCurrent()) == this) {
-               main->CloseTab(main->GetCurrent()); // emit signal
-               //main->RemoveTab(main->GetCurrent());
-               return kTRUE;
+      if (fParent && (event->fCode == kButton4 || event->fCode == kButton5)) { //scroll wheel events
+         TGTab* main = (TGTab*)fParent;
+
+         if(event->fCode == kButton4) { //scroll up = move left, as in Firefox
+            for(Int_t c = main->GetCurrent() - 1; c >= 0; --c) {
+               if(main->GetTabTab(c)->IsEnabled()) {
+                  // change tab and generate event
+                  main->SetTab(c);
+                  break;
+               }
+            }
+         } else if(event->fCode == kButton5) { //scroll down = move right, as in Firefox
+            for(Int_t c = main->GetCurrent() + 1; c < main->GetNumberOfTabs(); ++c) {
+               if(main->GetTabTab(c)->IsEnabled()) {
```

Comment:
Add a space between `if`, `for` and their corresponding opening parenthesis, like:  `if (` `for (`